DSpace/dspace-angular
# Fix search sidebar a11y issues

## References

* Fixes #1168

## Description

Addressed accessibility issues in the search sidebar

* Added labels
* Added `null` href values to all "empty" links for screen readers
* Improved keyboard navigation
  * Expand/collapse filters with keyboard; highlight facet when selected
  * Exclude controls in collapsed facets from the tablist

## Instructions for Reviewers

**Testing:**

1. Go to `/search`
2. Open some filter facets. 
   * [ ] Search boxes should have labels
   * [ ] Date range boxes should have labels
3. Navigate the sidebar using `Tab`/`Shift+Tab`
   * [ ] You should be able to expand/collapse facets with the keyboard
   * [ ] When a closed facet is selected, pressing `Tab` should select the next facet (old behaviour was to start selecting the hidden controls in the selected facet instead)

## Checklist

_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
